disnake/abc.py

```diff
@@ -630,6 +630,23 @@ def jump_url(self) -> str:
         """
         return f"https://discord.com/channels/{self.guild.id}/{self.id}"
 
+    def _apply_implict_permissions(self, base: Permissions) -> Permissions:
+        # if you can't send a message in a channel then you can't have certain
+        # permissions as well
+        if not base.send_messages:
+            base.send_tts_messages = False
+            base.send_voice_messages = False
+            base.mention_everyone = False
+            base.embed_links = False
+            base.attach_files = False
+
+        # if you can't view a channel then you have no permissions there
+        if not base.view_channel:
+            denied = Permissions.all_channel()
+            base.value &= ~denied.value
```

Comment:
could probably make this a bit "cleaner"(?) by reordering it to be
```py
if not base.view_channel:
    ...

elif not base.send_messages:
    ...
```
as it appears all of the permissions handled by the `send_messages` check would be set to False by the `view_channel` check anyways.

Reply:
as per comment on discord, this is out of scope of this issue. https://canary.discord.com/channels/808030843078836254/847513235536216085/1120096424072331384